src/notification.test.ts

```diff
@@ -437,7 +478,43 @@ describe("BaseNotification", () => {
       expect(notification.fetchProtocolNegotiationInfo).toHaveBeenCalled();
     });
 
-    test("throws a FetchError if the negotiation connection info fetch fails", async () => {
+    test("throws a BadRequestError if the negotiation connection info fetch fails, using problem details", async () => {
+      const fetchFn = mockedFetchWithError(
+        400,
+        `{"status": 400, "title": "Connection info error", "detail": "Connection info error detail"}`,
+        { "Content-Type": "application/problem+json" },
+      );
+
+      const endpoint = "https://fake.url/some-endpoint";
+      const topic = "https://fake.url/some-resource";
+      const protocol = ["ws"] as Array<protocols>;
+      const notification = new BaseNotification(topic, protocol, {
```

Comment:
```suggestion
      const notification = new BaseNotification(topic, ["ws"], {
```
The type assertion is unnecessary. I'm happy to keep the `protocol` variable declared if you find that more readable, but in any case you should remove `as Array<protocols>`.

Reply:
I removed the `as Array<protocols>` here and elsewhere in the code